File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/RiskEvidenceSynthesis.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import UsageContext
from base import Annotation
from base import Period
from base import ContactDetail
from base import CodeableConcept
from base import RelatedArtifact
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class RiskEvidenceSynthesisSampleSize(BackboneElement):
    description: Optional[str] = None
    number_of_studies: Optional[int] = None
    number_of_participants: Optional[int] = None

@dataclass(kw_only=True)
class RiskEvidenceSynthesisCertaintyCertaintySubcomponent(BackboneElement):
    note: Optional[List[Annotation]] = field(default_factory=list)
    type: Optional[CodeableConcept] = None
    rating: Optional[List[CodeableConcept]] = field(default_factory=list)

@dataclass(kw_only=True)
class RiskEvidenceSynthesisCertainty(BackboneElement):
    note: Optional[List[Annotation]] = field(default_factory=list)
    rating: Optional[List[CodeableConcept]] = field(default_factory=list)
    certainty_subcomponent: Optional[List[RiskEvidenceSynthesisCertaintyCertaintySubcomponent]] = field(default_factory=list)

@dataclass(kw_only=True)
class RiskEvidenceSynthesisRiskEstimatePrecisionEstimate(BackboneElement):
    to: Optional[float] = None
    from_: Optional[float] = None
    type: Optional[CodeableConcept] = None
    level: Optional[float] = None

@dataclass(kw_only=True)
class RiskEvidenceSynthesisRiskEstimate(BackboneElement):
    type: Optional[CodeableConcept] = None
    value: Optional[float] = None
    description: Optional[str] = None
    unit_of_measure: Optional[CodeableConcept] = None
    numerator_count: Optional[int] = None
    denominator_count: Optional[int] = None
    precision_estimate: Optional[List[RiskEvidenceSynthesisRiskEstimatePrecisionEstimate]] = field(default_factory=list)

@dataclass(kw_only=True)
class RiskEvidenceSynthesis(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    endorser: Optional[List[ContactDetail]] = field(default_factory=list)
    publisher: Optional[str] = None
    approval_date: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = field(default_factory=list)
    sample_size: Optional[RiskEvidenceSynthesisSampleSize] = None
    name: Optional[str] = None
    use_context: Optional[List[UsageContext]] = field(default_factory=list)
    copyright: Optional[str] = None
    study_type: Optional[CodeableConcept] = None
    outcome: Reference
    topic: Optional[List[CodeableConcept]] = field(default_factory=list)
    title: Optional[str] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    author: Optional[List[ContactDetail]] = field(default_factory=list)
    synthesis_type: Optional[CodeableConcept] = None
    status: str
    population: Reference
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    last_review_date: Optional[str] = None
    editor: Optional[List[ContactDetail]] = field(default_factory=list)
    certainty: Optional[List[RiskEvidenceSynthesisCertainty]] = field(default_factory=list)
    reviewer: Optional[List[ContactDetail]] = field(default_factory=list)
    exposure: Optional[Reference] = None
    version: Optional[str] = None
    related_artifact: Optional[List[RelatedArtifact]] = field(default_factory=list)
    contact: Optional[List[ContactDetail]] = field(default_factory=list)
    risk_estimate: Optional[RiskEvidenceSynthesisRiskEstimate] = None
    effective_period: Optional[Period] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/RiskEvidenceSynthesis.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class RiskEvidenceSynthesis : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? Date { get; set; }
    public Base.ContactDetail[]? Endorser { get; set; }
    public string? Publisher { get; set; }
    public string? ApprovalDate { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public RiskEvidenceSynthesis_SampleSize? SampleSize { get; set; }
    public string? Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public Base.CodeableConcept? StudyType { get; set; }
    public required Base.ResourceReference Outcome { get; set; }
    public Base.CodeableConcept[]? Topic { get; set; }
    public string? Title { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public Base.ContactDetail[]? Author { get; set; }
    public Base.CodeableConcept? SynthesisType { get; set; }
    public required string Status { get; set; }
    public required Base.ResourceReference Population { get; set; }
    public string? Url { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? LastReviewDate { get; set; }
    public Base.ContactDetail[]? Editor { get; set; }
    public RiskEvidenceSynthesis_Certainty[]? Certainty { get; set; }
    public Base.ContactDetail[]? Reviewer { get; set; }
    public Base.ResourceReference? Exposure { get; set; }
    public string? Version { get; set; }
    public Base.RelatedArtifact[]? RelatedArtifact { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }
    public RiskEvidenceSynthesis_RiskEstimate? RiskEstimate { get; set; }
    public Base.Period? EffectivePeriod { get; set; }

    public class RiskEvidenceSynthesis_SampleSize : BackboneElement
    {
        public string? Description { get; set; }
        public int? NumberOfStudies { get; set; }
        public int? NumberOfParticipants { get; set; }
    }

    public class RiskEvidenceSynthesis_Certainty_CertaintySubcomponent : BackboneElement
    {
        public Base.Annotation[]? Note { get; set; }
        public Base.CodeableConcept? Type { get; set; }
        public Base.CodeableConcept[]? Rating { get; set; }
    }

    public class RiskEvidenceSynthesis_Certainty : BackboneElement
    {
        public Base.Annotation[]? Note { get; set; }
        public Base.CodeableConcept[]? Rating { get; set; }
        public RiskEvidenceSynthesis_Certainty_CertaintySubcomponent[]? CertaintySubcomponent { get; set; }
    }

    public class RiskEvidenceSynthesis_RiskEstimate_PrecisionEstimate : BackboneElement
    {
        public string? To { get; set; }
        public string? From_ { get; set; }
        public Base.CodeableConcept? Type { get; set; }
        public string? Level { get; set; }
    }

    public class RiskEvidenceSynthesis_RiskEstimate : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public string? Value { get; set; }
        public string? Description { get; set; }
        public Base.CodeableConcept? UnitOfMeasure { get; set; }
        public int? NumeratorCount { get; set; }
        public int? DenominatorCount { get; set; }
        public RiskEvidenceSynthesis_RiskEstimate_PrecisionEstimate[]? PrecisionEstimate { get; set; }
    }
}